### PR TITLE
Ports instruments from /VG/station, including harmonicas, xylophones, saxophones, trombones, and a BIKE HORN. Repaths instruments to sound/instruments. Also, allows piano synth to play all instruments but eguitar and gives cargo a crate with everything. Clown gets the bike horn trumpet at roundstart.

### PR DESCRIPTION
**Pull request by silicons**
Title.
ok fuck that might of been too long a title.
~~also for those of you concerned with RSC size this replaces violin oggs with mids. Increase by almost 1 MB.~~
Did I say 1 MB bigger? My mistake.
Old
old
New
new

### CODE REVIEW

**Comment by Armhulen** [comment]
751 file changes for 5 new instruments :thinking:

**Comment by Armhulen** [comment]
oh i just checked code uuuuuuuh

**Comment by GunHog** [comment]
Kev, I am really worried right now...Kev...you made quite the mess here...Kev, stop using the online editor!

**Comment by meeeeme** [comment]
Synthesizer for pAI 2017!

**Comment by DoctoreMad** [comment]
>/VG/

Do you want to know why I'm laughing right now?

File: code/game/objects/items/devices/instruments.dm [hunk 1]
```diff
@@ -6,11 +6,10 @@
...
 	var/datum/song/handheld/song
 	var/instrumentId = "generic"
-	var/instrumentExt = "ogg"
+	var/instrumentExt = "mid"
 
 /obj/item/device/instrument/New()
-	song = new(instrumentId, src)
-	song.instrumentExt = instrumentExt
+	song = new(instrumentId, src, instrumentExt)
 	..()
```
Comment (Cyberboss) [inline code/game/objects/items/devices/instruments.dm]: Initialize and . = ..()

File: code/datums/action.dm [hunk 2]
```diff
@@ -236,6 +236,17 @@
...
 	..()
 
+/datum/action/item_action/synthswitch
+	name = "Change Synthesizer Instrument"
+	desc = "Change the type of instrument your synthesizer is playing as."
+
+/datum/action/item_action/synthswitch/Trigger()
+	if(istype(target, /obj/item/device/instrument/piano_synth))
+		var/obj/item/device/instrument/piano_synth/synth = target
+		synth.changeInstrument(input("Choose the type of instrument you want to use", "Instrument Selection", "piano") as null|anything in synth.insTypes)
```
Comment (Cyberboss) [inline code/datums/action.dm]: do sanity checks after the input

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by nicbn** [comment]
The increase in the game size doesn't compensate for the minimal advantages of having one septillion instruments.

**Comment by Cyberboss** [comment]
> 1MB

It's 2017, grow up

File: code/game/objects/items/devices/instruments.dm [hunk 3]
```diff
@@ -6,12 +6,11 @@
...
-	var/instrumentExt = "ogg"
+	var/instrumentExt = "mid"
 
-/obj/item/device/instrument/New()
-	song = new(instrumentId, src)
-	song.instrumentExt = instrumentExt
-	..()
+/obj/item/device/instrument/Initialize()
+	song = new(instrumentId, src, instrumentExt)
+	. = ..()
```
Comment (Cyberboss) [inline code/game/objects/items/devices/instruments.dm]: . = ..() first

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by octareenroon91** [comment]
>a file for every MIDI note-instrument combination

holy fuck

**Comment by Tacolizard** [comment]
WTF 1 MEGABYTE ON TOP OF MUH 1.1GB GAME?!?!?!?!?!!?!?!?!?!?!?!!?!?!?!?!?!?!?!?!?!?!?!?!?!??!?!?!?!?!?!?!?!??!?!?!!?!?!?!?!?!?!?!?!?!??!?!?!?!!?!?!!!?!?!?!?!?!?!?!??!!?!?!?!?!?!?!?!?!?!?!!?!?!?!?!?!?!!?!?!?!?!?!?!?!?!?!?!?!?!?!??!?!?!?!?!?!?!?!??!?!?!!?!?!?!?!?!?!?!?!?!??!?!?!?!!?!?!!!?!?!?!?!?!?!?!??!!?!?!?!?!?!?!?!?!?!?!!?!?!?!?!?!?!!?!?!?!?!?!?!?!?!?!?!?!?!?!??!?!?!?!?!?!?!?!??!?!?!!?!?!?!?!?!?!?!?!?!??!?!?!?!!?!?!!!?!?!?!?!?!?!?!??!!?!?!?!?!?!?!?!?!?!?!!?!?!?!?!?!?!!?!?!?!?!?!?!?!?!?!?!?!?!?!??!?!?!?!?!?!?!?!??!?!?!!?!?!?!?!?!?!?!?!?!??!?!?!?!!?!?!!!?!?!?!?!?!?!?!??!!?!?!?!?!?!?!?!?!?!?!!?!?!?!?!?!?!!?!?!?!?!?!?!?!?!?!?!?!?!?!??!?!?!?!?!?!?!?!??!?!?!!?!?!?!?!?!?!?!?!?!??!?!?!?!!?!?!!!?!?!?!?!?!?!?!??!!?!?!?!?!?!?!?!?!?!?!!?!?!?!?!?!?!!?!?!?!?!?!?!?!?!?!?!?!?!?!??!?!?!?!?!?!?!?!??!?!?!!?!?!?!?!?!?!?!?!?!??!?!?!?!!?!?!!!?!?!?!?!?!?!?!??!!?!?!?!?!?!?!?!?!?!?!!?!?!?!?!?!?!!?!?!?!?!?!?!?!?!?!?!?!?!?!??!?!?!?!?!?!?!?!??!?!?!!?!?!?!?!?!?!?!?!?!??!?!?!?!!?!?!!!?!?!?!?!?!?!?!??!!?!?!?!?!?!?!?!?!?!?!!?!?!?!?!?!?!!?!?!?!?!?!?!?!?!?!?!?!?!?!??!?!?!?!?!?!?!?!??!?!?!!?!?!?!?!?!?!?!?!?!??!?!?!?!!?!?!!!?!?!?!?!?!?!?!??!!?!?!?!?!?!?!?!?!?!?!!?!?!?!?!?!?!!?!?!?!?!?!?!?!?!?!?!?!?!?!??!?!?!?!?!?!?!?!??!?!?!!?!?!?!?!?!?!?!?!?!??!?!?!?!!?!?!!!?!?!?!?!?!?!?!??!!?!?!?!?!?!

**Comment by kevinz000** [comment]
@Tacolizard 58 mb or something is how much players download iirc

**Comment by PKPenguin321** [comment]
Why can't synth play eguitar

**Comment by octareenroon91** [comment]
>Why can't synth play eguitar

That's an intentional choice. Maybe something about ear-rape.

**Comment by optimumtact** [comment]
It's easy for you americans to not complain about the filesize, but it can take almost 5-6 minutes to download the tgstation resources if the CDN fails and or byond decides to invalidate the game cache.

**Comment by CPTANT** [comment]
750 extra files and 1MB of extra gamesize for 5 instruments?

No thank you.

**Comment by kevinz000** [comment]
@PKPenguin321 i don't want to add more eguitar instruments, they're earrapey as fuck, loud, and obnoxious. if they want them there's one on the station roundstart on most maps I believe.

**Comment by kevinz000** [comment]
@CPTANT 
1. The only thing you should be worried about is gamesize. Number of files isn't really a concern as I don't believe byond actually makes 750 files on your computer, but if I'm wrong it still isn't that bad.
2. Many of the 750 files is from me moving all current instruments to another folder.

**Comment by kevinz000** [comment]
Did I say 1 MB bigger? My mistake.
Old
old
New
new

**Comment by kevinz000** [comment]
@PKPenguin321 il add eguitar to it in a later pr that also lowers the eguitar sounds, as well as eguitars to the crate because jesus christ they're too loud..

**Comment by PraiseRatvar** [comment]
why are people so pissy over like 3 megabytes

**Review by Cyberboss — APPROVED** [review]

**Comment by Cyberboss** [comment]
>actually decreases file size

Shows all it takes to start a shitstorm around here

**Comment by nicbn** [comment]
> WTF 1 MEGABYTE ON TOP OF MUH 1.1GB GAME?!?!?
Not about me, it's about the players who already download around 50 MB to play.
> It's 2017, grow up
I live in Brazil. A third world country. My internet is one of the few good there are, but until december of the last year I had a 2 Mbps internet. Actually internet providers lie, so it was 1 Mbps, sometimes 0.5 Mbps.
Arount 7 minutes to play tg. That was one of the reasons I played tg instead of para. Para was 15 minutes to download. 15 fucking minutes. That's why I ask you: Don't take a step in this direction. Now it already has been taken, but please, when adding more assets, think about people who need to wait to play because they live in third world countries, and, in case of Brazil, countries where things normally don't work for them.

**Comment by Cyberboss** [comment]
It's a half a megabyte DECREASE you memes

**Comment by nicbn** [comment]
It's not anymore because of the file size, it's because you was rude about me complaining it. Come on, not everybody likes to have little more features - that they will probably never use - in tradeoff of more content to download.
If you showed the file decrease instead of saying me to "grow up" I'd be totally fine, but come on, not everybody lives in USA/Europe.

**Comment by Cyberboss** [comment]
Funny how we get these complaints now after rsc size has dropped ~7MB in the past month, but w/e

**Comment by optimumtact** [comment]
@kevinz000 wrong again, considering that often another file requires a new tcp request from the client, which entails the full tcp overhead of setting up a connection.

Especially a big issue on the webclient

**Comment by optimumtact** [comment]
Err that was about file count

**Comment by Fox-McCloud** [comment]
@kevinz000 

We actually made a PR on Para to lower their volume: https://github.com/ParadiseSS13/Paradise/pull/7768

they're very reasonable and in-line with the guitar, piano, and violin.

Also, you didn't delete the old instrument folders....andddd you used the shitty midi violin as opposed to the .ogg version.

Also, you added xylophone midis and sprites, but didn't include it as an instrument??

**Pull request merged**